### DISCUSSION TITLE
Changing model settings to Sentence Bert

### DISCUSSION THREAD

**Discussion by metliv** [asreview/asreview#1674] [Q&A] [answered]
Hi, we are doing the SAFE methode and are in phase 3 now.  Is there any way to change the model settings in the project? We wanted to use as a feature extraction technique: Sentence Bert and for the Classfier: fully connected neutral network (2 hidden layers) and we exported a ris-file of the porject with the labeled and und unlabeled records and created a new project with the mentioned settings. We also performed "pip install asreview[all]" to insatll SBert model configuration. I can create the project but when I start reviewig I get the notification "Failed to retrieve new records. Save trained model data before using this function.." How can I fix the problem?

**Comment by jteijema** [comment]
Hi @metliv ,

Changing to sbert and using the NN-2-Layer is a great idea! Let's try to figure out the root of your issue.

As of right now, changing model during the review is not possible, therefore exporting your data and reimporting was the correct choice. Can you run, in the console, the command `pip freeze` for me? That will help me see if the installation worked correctly for the use of those models. It's an error I haven't seen before!

^JT

**Reply by metliv** [reply]
Hi @jteijema ,
thank you for your help. I tried again to set up the project and now it's not even creating the project with the mentioned settings anymore. If I run pip freeze I get this output:

absl-py==2.1.0
asreview==1.5
asreview-datatools==1.1.1
astunparse==1.6.3
attrs==23.1.0
blinker==1.6.2
build==1.0.3
cachetools==5.3.2
certifi==2023.7.22
cffi==1.15.1
charset-normalizer==3.2.0
check-manifest==0.49
click==8.1.6
colorama==0.4.6
datahugger==0.3
decorator==5.1.1
dill==0.3.7
et-xmlfile==1.1.0
filelock==3.12.2
Flask==2.3.2
Flask-Cors==4.0.0
Flask-Login==0.6.3
Flask-Mail==0.9.1
Flask-SQLAlchemy==3.1.1
flatbuffers==23.5.26
fsspec==2024.2.0
gast==0.4.0
gensim==4.3.2
gevent==23.7.0
gitignorefile==1.1.2
google-auth==2.27.0
google-auth-oauthlib==1.0.0
google-pasta==0.2.0
greenlet==2.0.2
grpcio==1.60.1
h5py==3.10.0
huggingface-hub==0.20.3
idna==3.4
itsdangerous==2.1.2
jax==0.4.23
Jinja2==3.1.2
joblib==1.3.1
jsonpath-ng==1.5.3
jsonschema==4.18.4
jsonschema-specifications==2023.7.1
keras==2.12.0
libclang==16.0.6
Markdown==3.5.2
MarkupSafe==2.1.3
ml-dtypes==0.3.2
mpmath==1.3.0
natsort==8.4.0
networkx==3.2.1
nltk==3.8.1
numpy==1.24.3
oauthlib==3.2.2
openpyxl==3.1.2
opt-einsum==3.3.0
packaging==23.2
pandas==2.0.3
pillow==10.2.0
ply==3.11
protobuf==4.25.2
pyalex==0.13
pyasn1==0.5.1
pyasn1-modules==0.3.0
pycparser==2.21
pyproject_hooks==1.0.0
python-dateutil==2.8.2
pytz==2023.3
PyYAML==6.0.1
referencing==0.30.0
regex==2023.12.25
requests==2.31.0
requests-oauthlib==1.3.1
rispy==0.7.1
rpds-py==0.9.2
rsa==4.9
safetensors==0.4.2
scikit-learn==1.3.0
scipy==1.11.1
scisort==0.5.3
scitree==0.5.3
seedir==0.4.2
sentence-transformers==2.3.1
sentencepiece==0.1.99
six==1.16.0
smart-open==6.4.0
SQLAlchemy==2.0.24
SQLAlchemy-Utils==0.41.1
sympy==1.12
synergy-dataset==1.2
tabulate==0.9.0
tensorboard==2.12.3
tensorboard-data-server==0.7.2
tensorflow==2.12.1
tensorflow-estimator==2.12.0
tensorflow-intel==2.12.1
tensorflow-io-gcs-filesystem==0.31.0
termcolor==2.4.0
threadpoolctl==3.2.0
tokenizers==0.15.1
torch==2.1.2
tqdm==4.65.0
transformers==4.37.2
typing_extensions==4.5.0
tzdata==2023.3
urllib3==2.0.4
Werkzeug==2.3.6
wrapt==1.14.1
xlrd==2.0.1
XlsxWriter==3.1.9
zope.event==5.0
zope.interface==6.0

My colleague has the same problem and she gets this output after running pip freeze:
asreview==1.2.1
attrs==23.1.0
blinker==1.6.3
cattrs==23.1.2
certifi==2023.7.22
cffi==1.16.0
charset-normalizer==3.3.0
click==8.1.7
colorama==0.4.6
datahugger==0.10.3
dill==0.3.7
et-xmlfile==1.1.0
filelock==3.12.4
Flask==3.0.0
Flask-Cors==4.0.0
gevent==23.9.1
gitignorefile==1.1.2
greenlet==3.0.0
idna==3.4
itsdangerous==2.1.2
Jinja2==3.1.2
joblib==1.3.2
jsonpath-ng==1.6.0
jsonschema==4.19.1
jsonschema-specifications==2023.7.1
MarkupSafe==2.1.3
natsort==8.4.0
numpy==1.26.1
openpyxl==3.1.2
pandas==2.1.1
platformdirs==3.11.0
ply==3.11
pycparser==2.21
python-dateutil==2.8.2
pytz==2023.3.post1
referencing==0.30.2
requests==2.31.0
requests-cache==1.1.0
rispy==0.7.1
rpds-py==0.10.6
scikit-learn==1.3.1
scipy==1.11.3
scisort==0.5.3
scitree==0.5.3
seedir==0.4.2
setuptools==68.2.2
six==1.16.0
threadpoolctl==3.2.0
tqdm==4.66.1
tzdata==2023.3
url-normalize==1.4.3
urllib3==2.0.6
Werkzeug==3.0.0
xlrd==2.0.1
zope.event==5.0
zope.interface==6.1

**Reply by jteijema** [reply]
Alright. Looks like the installation of packages is correct. Then it must be the data or the python installation itself. Is the data something you could share? Otherwise email me at J.j.teijema@uu.nl. I will share the solution on the forum for future reference.

**Reply by jteijema** [reply]
We've established that the exported RIS file works on my device using sbert, leaving the assumption that the system you are processing files on isn't cooperating.

**Reply by NAli42** [reply]
Hi! I am also doing the SAFE procedure and am on the SBERT step. However, when I try to use the model I get an error that sentence-transformers needs to be installed and when I follow the instructions to install all dependencies ASReview doesn't open anymore. 
The error I get is  "The specified module could not be found. Error loading "C:\...\Python\Python312\Lib\site-packages\torch\lib\fbgemm.dll" or one of its dependencies.".
I tried uninstalling torch and ASReview does open then, but of course I can't actually use the model then. Any idea what is going wrong?
Thanks in advance!

**Reply by jteijema** [reply, marked as answer]
I can confirm @NAli42 's issue. 

![image](https://github.com/user-attachments/assets/ed87e086-0197-4b73-a507-76c730618c9c)

This is a known issue with the torch package on python 3.12 (and not an issue with ASReview). 

Here is a stack overflow thread on the issue from a week ago:
https://stackoverflow.com/questions/78786306/fbgemm-load-error-trying-to-use-pytorch-on-windows

The following command was the solution for me:

```bash
pip uninstall torch

pip install torch==2.3.0 tf-keras
```

**Reply by NAli42** [reply]
That worked, thanks a lot for the quick solution!
